components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/ApiModelToOAuthConsumerApp.java

```diff
@@ -178,4 +188,46 @@ private String getCallbackUrl(List<String> callbackURLs) {
             return null;
         }
     }
+
+    private void updateClientAuthenticationConfigurations(OAuthConsumerAppDTO appDTO,
+                                                          ClientAuthenticationConfiguration clientAuthentication) {
+
+        if (clientAuthentication != null) {
+            appDTO.setTokenEndpointAuthMethod(clientAuthentication.getTokenEndpointAuthMethod());
+            appDTO.setTokenEndpointAuthSignatureAlgorithm(clientAuthentication.getTokenEndpointAuthSigningAlg());
+            appDTO.setTlsClientAuthSubjectDN(clientAuthentication.getTlsClientAuthSubjectDn());
+        }
+    }
+
+    private void updatePARConfigurations(OAuthConsumerAppDTO appDTO, PushAuthorizationRequestConfiguration par) {
+
+        if (par != null) {
+            appDTO.setRequirePushedAuthorizationRequests(par.getRequirePushAuthorizationRequest());
+        }
+    }
+
+    private void updateRequestObjectConfiguration(OAuthConsumerAppDTO consumerAppDTO,
+                                                  RequestObjectConfiguration requestObject) {
+
+        if (requestObject != null) {
+
+            if (requestObject.getEncryption().getAlgorithm() != null) {
```

Comment:
need to null check for getEncryption() as if request is sent without adding any encryption configs the encryption obj will be null. Fix other relevant places as well.
```suggestion
            if (requestObject.getEncryption() !=  null && requestObject.getEncryption().getAlgorithm() != null) {
```

Reply:
Fixed in 5285513bf8257aa1ae3c79093209271a18f36da1